### ISSUE TITLE
Can't connect to kubernetes from WSL

### ISSUE THREAD

**Issue by maxisam** [docker/for-win#13271]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID:

### Actual behavior

- Enable kubernetes
- try to connect to kubernetes but fail with 
"The connection to the server kubernetes.docker.internal:6443 was refused - did you specify the right host or port?"

It works in powershell. And it was working before until today I hit the reset button for kubernetes in docker desktop. (or after update?)

It can't connect to kubernetes anymore. I check the configuration file under .kube folder in Linux with the one in Windows

I have tried --kubeconfig. That doesn't work either.

They are identical. I also use another WSL distro and it doesn't work either. 

I can ping kubernetes.docker.internal and it returns 127.0.0.1

WSL integration is enabled

### Expected behavior

It should work. 

### Information

  - Windows Version:

WSL version: 1.1.3.0
Kernel version: 5.15.90.1
WSLg version: 1.0.49
MSRDC version: 1.2.3770
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22000.1574

  - Docker Desktop Version:

Docker Desktop 4.17.0 (99724)

  - WSL2 or Hyper-V backend?  

WSL2

  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM:

NO

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`
```
[2023-03-01T21:50:30.948354200Z][com.docker.diagnose.exe][I] set path configuration to OnHost
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0002: does the bootloader have virtualization enabled?
[SKIP] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[FAIL] DD0029: is the WSL 2 Linux filesystem corrupt? [    7.477273] EXT4-fs error (device sdd): ext4_put_super:1188: comm Xwayland: Couldn't clean up the journal
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0005: is the user in the docker-users group?
[PASS] DD0038: is the connection to Docker working?
[PASS] DD0014: are the backend processes running?
[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0006: is the Docker Desktop Service responding?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[WARN] DD0029: is the WSL 2 Linux filesystem corrupt? [    7.477273] EXT4-fs error (device sdd): ext4_put_super:1188: comm Xwayland: Couldn't clean up the journal
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0032: do Docker networks overlap with host IPs?

```

### Steps to reproduce the behavior
1. enable kubernetes from docker desktop and hit reset button
2. can't connect to it under WSL

**Comment by maxisam** [comment]
Another interesting thing is I can't install nginx anymore

helm upgrade --install ingress-nginx ingress-nginx --repo https://kubernetes.github.io/ingress-nginx --namespace ingress-nginx --create-namespace

It throws something like

2023/03/01 22:17:20 [emerg] 31#31: bind() to 0.0.0.0:443 failed (98: Address in use)

**Comment by StefanScherer** [comment]
Hello @maxisam Thanks for reporting this.
I've tried the same and can't reproduce the problem with DD 4.17.0 on my machine. Also the helm command works in my WSL distro.
The `is the WSL 2 Linux filesystem corrupt?` failure/warning looks suspicious.
I suggest to do a Factory Reset in DD and enable Kubernetes again. Does it work after that?

**Comment by maxisam** [comment]
I figured out the helm issue. I think windows ip helper service caused it, since it use port 80 and 443. 

I will try Factory Reset in DD later today. Also DD crash with Visual Stuido bridge to kubernetes, but Minikube works fine.

**Comment by manklu** [comment]
Since today I have the same problem.
Nobody in the WSL distro is listening for port 6443.

`sudo netstat -ltnp

Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp6       0      0 :::443                  :::*                    LISTEN      -
tcp6       0      0 :::80                   :::*                    LISTEN      -
tcp6       0      0 :::30000                :::*                    LISTEN      -
tcp6       0      0 :::4447                 :::*                    LISTEN      -
tcp6       0      0 :::4448                 :::*                    LISTEN      -
`
Ports for accessing services are mapped correctly.

**Comment by maxisam** [comment]
So I try reset, that one doesn't work. It got stuck in the process. In the end, I remove DD, manually remove all files in .docker and reinstall it. (There was an error saying can't connect to wsl but after restarting DD, everything seems like working )

However, it still doesn't work. I still can't connect to kubernetes in wsl but it works fine under windows


but I got everything green now

```
[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0002: does the bootloader have virtualization enabled?
[SKIP] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0005: is the user in the docker-users group?
[PASS] DD0038: is the connection to Docker working?
[PASS] DD0014: are the backend processes running?
[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0006: is the Docker Desktop Service responding?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0032: do Docker networks overlap with host IPs?
No fatal errors detected.
```

**Comment by manklu** [comment]
In .wslconfig I have set localhostForwarding to false and it works again.

It seems to be a problem in combination with this feature.

**Comment by maxisam** [comment]
So the way I fixed mine is update everything to the latest version.

WSL version: 1.1.3.0
Kernel version: 5.15.90.1
WSLg version: 1.0.49
MSRDC version: 1.2.3770
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.1344


I have tried removing and reinstalling wsl / linux subsystem / docker. None of them work. 

So in the end, I upgrade Windows to 22H2 and that fixed this issue. 

I think it is less likely DD's kubernetes issue. I think it is more like WSL2 network somehow doesn't work properly.

I was testing "bridge to kubernetes" before this issue happened. Maybe that thing change some network settings.
